### PR TITLE
build(deps): bump github.com/CosmWasm/wasmvm to v1.0.0-beta10

### PR DESCRIPTION
**Pull request by bragaz**
## Description

Closes: #791
Closes: #811   

In order to correctly update it, I've also updated our:
- cosmos-sdk fork (https://github.com/desmos-labs/cosmos-sdk/commit/71d2ba21554f5884353751856dff69ad312eeb15)
- wasmd fork (https://github.com/desmos-labs/wasmd/commit/011bbf16d479584e7f92010cefe5dcd2a61074fc)

Please @RiccardoM take a look also at the cosmos-sdk forks, since I didn't know what was edited there, maybe you will see a better way to updated it. 

Depends-On: #816
Depends-On: #817

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://docs.cosmos.network/v0.44/building-modules/intro.html)
- [ ] included the necessary unit and integration [tests](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by bragaz** [comment]
Can you reproduce the lint error on your hardware? @RiccardoM @dadamu @manu0466 , cause I don't see it on my Mac

**Comment by manu0466** [comment]
@bragaz I tried with 2 versions 
* **v1.28**: `docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:v1.28 golangci-lint run -v` and i get this error: 
```
level=warning msg="[runner] Can't run linter goanalysis_metalinter: bodyclose: failed prerequisites: [buildssa@github.com/desmos-labs/desmos/v3/x/oracle/types: analysis skipped: errors in package: [/app/x/oracle/types/codec.go:4:2: could not import github.com/cosmos/cosmos-sdk/codec (/go/pkg/mod/github.com/desmos-labs/cosmos-sdk@v0.43.0-alpha1.0.20220407145510-71d2ba21554f/codec/amino.go:11:10: could not import github.com/tendermint/tendermint/types (/go/pkg/mod/github.com/tendermint/tendermint@v0.34.16/types/event_bus.go:7:2: could not import github.com/tendermint/tendermint/abci/types (/go/pkg/mod/github.com/tendermint/tendermint@v0.34.16/abci/types/types.pb.go:15:7: could not import google.golang.org/grpc (/go/pkg/mod/google.golang.org/grpc@v1.33.2/clientconn.go:42:2: could not import google.golang.org/grpc/internal/transport (/go/pkg/mod/google.golang.org/grpc@v1.33.2/internal/transport/controlbuf.go:28:2: could not import golang.org/x/net/http2 (/go/pkg/mod/golang.org/x/net@v0.0.0-20211208012354-db4efeb81f4b/http2/transport.go:417:48: ErrDeadlineExceeded not declared by package os)))))) /app/x/oracle/types/codec.go:5:13: could not import github.com/cosmos/cosmos-sdk/codec/types (/go/pkg/mod/github.com/desmos-labs/cosmos-sdk@v0.43.0-alpha1.0.20220407145510-71d2ba21554f/codec/types/any.go:8:12: could not import github.com/cosmos/cosmos-sdk/types/errors (/go/pkg/mod/github.com/desmos-labs/cosmos-sdk@v0.43.0-alpha1.0.20220407145510-71d2ba21554f/types/errors/abci.go:7:7: could not import github.com/tendermint/tendermint/abci/types (/go/pkg/mod/github.com/tendermint/tendermint@v0.34.16/abci/types/types.pb.go:15:7: could not import google.golang.org/grpc (/go/pkg/mod/google.golang.org/grpc@v1.33.2/clientconn.go:42:2: could not import google.golang.org/grpc/internal/transport (/go/pkg/mod/google.golang.org/grpc@v1.33.2/internal/transport/controlbuf.go:28:2: could not import golang.org/x/net/http2 (/go/pkg/mod/golang.org/x/net@v0.0.0-20211208012354-db4efeb81f4b/http2/transport.go:417:48: ErrDeadlineExceeded not declared by package os)))))) /app/x/oracle/types/error.go:4:12: could not import github.com/cosmos/cosmos-sdk/types/errors (/go/pkg/mod/github.com/desmos-labs/cosmos-sdk@v0.43.0-alpha1.0.20220407145510-71d2ba21554f/types/errors/abci.go:7:7: could not import github.com/tendermint/tendermint/abci/types (/go/pkg/mod/github.com/tendermint/tendermint@v0.34.16/abci/types/types.pb.go:15:7: could not import google.golang.org/grpc (/go/pkg/mod/google.golang.org/grpc@v1.33.2/clientconn.go:42:2: could not import google.golang.org/grpc/internal/transport (/go/pkg/mod/google.golang.org/grpc@v1.33.2/internal/transport/controlbuf.go:28:2: could not import golang.org/x/net/http2 (/go/pkg/mod/golang.org/x/net@v0.0.0-20211208012354-db4efeb81f4b/http2/transport.go:417:48: ErrDeadlineExceeded not declared by package os))))) /app/x/oracle/types/keys.go:4:6: could not import github.com/cosmos/cosmos-sdk/types (/go/pkg/mod/github.com/desmos-labs/cosmos-sdk@v0.43.0-alpha1.0.20220407145510-71d2ba21554f/types/abci.go:3:13: could not import github.com/tendermint/tendermint/abci/types (/go/pkg/mod/github.com/tendermint/tendermint@v0.34.16/abci/types/types.pb.go:15:7: could not import google.golang.org/grpc (/go/pkg/mod/google.golang.org/grpc@v1.33.2/clientconn.go:42:2: could not import google.golang.org/grpc/internal/transport (/go/pkg/mod/google.golang.org/grpc@v1.33.2/internal/transport/controlbuf.go:28:2: could not import golang.org/x/net/http2 (/go/pkg/mod/golang.org/x/net@v0.0.0-20211208012354-db4efeb81f4b/http2/transport.go:417:48: ErrDeadlineExceeded not declared by package os))))) /app/x/oracle/types/oracle.pb.go:11:2: \"github.com/cosmos/cosmos-sdk/types\" imported but not used]]"
level=info msg="[linters context/goanalysis] analyzers took 54.446218ms with top 10 stages: buildir: 42.46532ms, U1000: 11.980898ms"
level=warning msg="[runner] Can't run linter unused: buildir: failed to load package http2: could not load export data: no export data for \"golang.org/x/net/http2\""
level=info msg="[runner] processing took 2.727µs with stages: max_same_issues: 545ns, nolint: 292ns, skip_dirs: 258ns, path_prefixer: 247ns, max_from_linter: 186ns, cgo: 130ns, autogenerated_exclude: 120ns, skip_files: 119ns, path_prettifier: 118ns, diff: 116ns, identifier_marker: 115ns, filename_unadjuster: 114ns, source_code: 62ns, exclude-rules: 59ns, max_per_file_from_linter: 50ns, path_shortener: 50ns, exclude: 49ns, severity-rules: 49ns, uniq_by_line: 48ns"
level=info msg="[runner] linters took 7.689614991s with stages: goanalysis_metalinter: 7.241551248s, unused: 448.00908ms"
level=error msg="Running error: buildir: failed to load package http2: could not load export data: no export data for \"golang.org/x/net/http2\""
```
* **v1.45.2**: `docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:v1.45.2 golangci-lint run -v` and i get this error:
```
level=info msg="[runner] linters took 16.88666515s with stages: goanalysis_metalinter: 16.858166111s"
x/oracle/types/keys.go:94:9: appendAssign: append result not assigned to the same slice (gocritic)
        buf := append(ReportStoreKeyPrefix, sdk.Uint64ToBigEndian(uint64(reqID))...)
               ^
x/profiles/keeper/grpc_query.go:142:20: appendAssign: append result not assigned to the same slice (gocritic)
                keyWithPrefix := append(ownersPrefix, key...)
                                 ^
x/profiles/keeper/grpc_query.go:230:20: appendAssign: append result not assigned to the same slice (gocritic)
                keyWithPrefix := append(ownersPrefix, key...)
                                 ^
app/desmos/cmd/testnet.go:395:15: badCall: suspicious Join on 1 argument (gocritic)
        writePath := filepath.Join(dir)
                     ^
x/profiles/simulation/operations.go:21:2: G101: Potential hardcoded credentials (gosec)
        OpWeightMsgSaveProfile         = "op_weight_msg_save_profile"
        ^
x/profiles/simulation/operations.go:22:2: G101: Potential hardcoded credentials (gosec)
        OpWeightMsgDeleteProfile       = "op_weight_msg_delete_profile"
        ^
x/profiles/simulation/operations.go:23:2: G101: Potential hardcoded credentials (gosec)
        OpWeightMsgRequestDTagTransfer = "op_weight_msg_request_dtag_transfer"
        ^
x/profiles/simulation/operations.go:24:2: G101: Potential hardcoded credentials (gosec)
        OpWeightMsgAcceptDTagTransfer  = "op_weight_msg_accept_dtag_transfer_request"
        ^
x/profiles/simulation/operations.go:25:2: G101: Potential hardcoded credentials (gosec)
        OpWeightMsgRefuseDTagTransfer  = "op_weight_msg_refuse_dtag_transfer_request"
        ^
x/profiles/simulation/operations.go:26:2: G101: Potential hardcoded credentials (gosec)
        OpWeightMsgCancelDTagTransfer  = "op_weight_msg_cancel_dtag_transfer_request"
        ^
x/relationships/simulation/operations.go:23:2: G101: Potential hardcoded credentials (gosec)
        OpWeightMsgCreateRelationship = "op_weight_msg_create_relationship"
        ^
x/relationships/simulation/operations.go:24:2: G101: Potential hardcoded credentials (gosec)
        OpWeightMsgDeleteRelationship = "op_weight_msg_delete_relationship"
        ^
x/relationships/simulation/operations.go:25:2: G101: Potential hardcoded credentials (gosec)
        OpWeightMsgBlockUser          = "op_weight_msg_block_user"
        ^
x/relationships/simulation/operations.go:26:2: G101: Potential hardcoded credentials (gosec)
        OpWeightMsgUnBlockUser        = "op_weight_msg_unblock_user"
        ^
testutil/ibctesting/chain.go:556:5: SA5011(related information): this check suggests that the pointer can be nil (staticcheck)
        if tmValSet != nil {
           ^
x/profiles/types/msgs_app_links.go:14:1: directive `// nolint:interfacer` is unused for linter "interfacer" (nolintlint)
// nolint:interfacer
^
x/profiles/types/msgs_app_links.go:87:1: directive `// nolint:interfacer` is unused for linter "interfacer" (nolintlint)
// nolint:interfacer
^
x/profiles/simulation/utils.go:76:1: directive `// nolint:interfacer` is unused for linter "interfacer" (nolintlint)
// nolint:interfacer
^
```
```

**Comment by RiccardoM** [comment]
There was no need to update the Cosmos SDK fork as well since the `desmos-labs/wasmd@v0.45.x-fixes` already contained the adapted code. I've now moved the most up-to-date code to [`desmos-labs/wasmd@desmos/master-fixes`](https://github.com/desmos-labs/wasmd) to make sure we only have one branch that needs to be maintained in order to be compatible with our Cosmos SDK fork. I've updated those references here as well. Lint is passing as well as all other tests

**Review by RiccardoM — APPROVED** [review]
The lint error was due to a change inside [`golang-ci-lint-action@v3`: ](https://github.com/golangci/golangci-lint-action#compatibility): 

> - `v3.0.0+` requires explicit setup-go installation step prior to using this action: `uses: actions/setup-go@v2`.
>   The `skip-go-installation` option has been removed.

After adding that step to the GitHub action, everything was solved properly :+1: 

**Note**. Tests are failing because there are some failing tests inside the `master`, nothing related to CosmWASM in particular.

**Comment by bragaz** [comment]
@RiccardoM Great, thanks for the heads up!

File: go.mod [hunk 1]
```diff
@@ -9,7 +9,7 @@ require (
 	github.com/cosmos/cosmos-sdk v0.45.1
 	github.com/cosmos/go-bip39 v1.0.0
 	github.com/cosmos/ibc-go/v2 v2.2.0
-	github.com/ethereum/go-ethereum v1.10.17
+	github.com/ethereum/go-ethereum v1.10.16
```
Comment (dadamu) [inline go.mod:12]:
```suggestion
	github.com/ethereum/go-ethereum v1.10.17
```
It should be updated to v1.10.17 already by https://github.com/desmos-labs/desmos/pull/799.

**Pull request merged**